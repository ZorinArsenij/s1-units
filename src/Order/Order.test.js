import React from 'react';
import Order from './Order';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

const defaultProps = {
    shop: '1', 
    data: 1
}

configure({ adapter: new Adapter() });

describe('Order component', () => {
  const wrapper = shallow(<Order order={defaultProps}/>);

  it('shop name render', () => {
    const order = {
        shop: 'shop',
        date: 1,
        items: [
            'item1',
            'item2',
            'item3',
        ]
    };

    wrapper.setProps({order: order})
    expect(wrapper.find(".Order-shop").contains(order.shop)).toBeTruthy();
  });

  it('has items', () => {
    const order = {
        shop: 'shop',
        date: 1,
        items: [
            'item1',
            'item2',
            'item3',
        ]
    };

    wrapper.setProps({order: order})
    expect(wrapper.find(".Order-items").children()).toHaveLength(order.items.length)
  });

  it('has not items', () => {
    const order = {
        shop: 'shop',
        date: 1,
    };

    wrapper.setProps({order: order})
    expect(wrapper.find(".Order-items").children()).toHaveLength(0)
  });
});
