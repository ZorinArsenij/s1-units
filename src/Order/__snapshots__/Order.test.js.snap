// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component render with default state DATE 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span>
      1 января, чт, 1970 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      item1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      item2
    </div>
    <div
      className="Order-item"
      key="2"
    >
      item3
    </div>
  </div>
</div>
`;
